string dir_path
---
bool ok
int32 processed

